pkg/appender/appender.go

```diff
@@ -215,20 +220,32 @@ func (mc *MetricsCache) Add(lset utils.LabelsIfc, t int64, v interface{}) (uint6
 // fast Add to metric (by refId)
 func (mc *MetricsCache) AddFast(ref uint64, t int64, v interface{}) error {
 
+	err := verifyTimeValid(t)
+	if err != nil {
+		return err
+	}
+
 	metric, ok := mc.getMetricByRef(ref)
 	if !ok {
 		mc.logger.ErrorWith("Ref not found", "ref", ref)
 		return fmt.Errorf("ref not found")
 	}
 
-	err := metric.error()
+	err = metric.error()
 	metric.setError(nil)
 
 	mc.appendTV(metric, t, v)
 
 	return err
 }
 
+func verifyTimeValid(t int64) error {
+	if t > 13569465600000 || t < 20000000000 {
```

Comment:
consider using human readable time format for better readability
Example: https://stackoverflow.com/questions/20924303/date-time-comparison-in-golang